internal/providers/kubernetes/provider.go

```diff
@@ -47,15 +45,15 @@ func New(client k8s.CoreV1Client, defaultMode string, meshNamespace string, tcpS
 func (p *Provider) buildRouter(name, namespace, ip string, port int, serviceName string, addMiddlewares bool) *dynamic.Router {
 	if addMiddlewares {
 		return &dynamic.Router{
-			Rule:        fmt.Sprintf("Host(`%s.%s.%s`) || Host(`%s`)", name, namespace, p.meshNamespace, ip),
+			Rule:        fmt.Sprintf("Host(`%s.%s.maesh`) || Host(`%s`)", name, namespace, ip),
```

Comment:
Does that work with the hardcoded .maesh here? I'm unsure, I can be totaly wrong tough!

Reply:
Hmm, from what I saw, `.maesh` is already hardcoded [here](https://github.com/containous/maesh/blob/master/internal/k8s/client.go#L365).

Reply:
This is the TLD. It is hardcoded for now. 😄

Reply:
This is the TLD. It is hardcoded for now.  😄 we effectively hardcoded it earlier by mandating that the `maeshNamespace` == "maesh".

Now it is actually hardcoded 🔨

Reply:
Wait I'm a bit late to the party but isn't hardcoding `maeshNamespace` to `maesh` kind of the opposite of the goal of this PR? 🤔

Reply:
Not really. The hardcoded `.maesh`  is already something enforced by the DNS rewrites rules we set up. 

The goal of this PR is to make maesh ignores it's resources in the namespace it's running on.